server/backend/rpcserver.go

```diff
@@ -34,7 +43,36 @@ func New(db *gorm.DB) *CampusServer {
 	}
 }
 
-func (s *CampusServer) GetTopNews(ctx context.Context, in *pb.GetTopNewsRequest) (*pb.GetTopNewsReply, error) {
+func (s *CampusServer) GetNewsSources(ctx context.Context, _ *emptypb.Empty) (newsSources *pb.NewsSourceArray, err error) {
+	if err = s.checkDevice(ctx); err != nil {
+		return
```

Comment:
is this new syntax that you no longer need to add `nil, err` here? :open_mouth:

Reply:
I don't think it's super new, you can omit named return values from the return statement and implicitly return whatever is in those vars 👌

Reply:
funky, not a fan, but if it works...